presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveFileSystem.java

```diff
@@ -249,7 +251,17 @@ protected void setup(String host, int port, String databaseName, BiFunction<Hive
                 new HiveWriterStats(),
                 getDefaultOrcFileWriterFactory(config, metastoreClientConfig),
                 columnConverterProvider);
-        pageSourceProvider = new HivePageSourceProvider(config, hdfsEnvironment, getDefaultHiveRecordCursorProvider(config, metastoreClientConfig), getDefaultHiveBatchPageSourceFactories(config, metastoreClientConfig), getDefaultHiveSelectivePageSourceFactories(config, metastoreClientConfig), FUNCTION_AND_TYPE_MANAGER, ROW_EXPRESSION_SERVICE);
+        Set<HiveRecordCursorProvider> recordCursorProviderSet = s3SelectPushdownEnabled ?
```

Comment:
This seemed a little strange at first, since I assumed `s3SelectPushdownEnabled` would provide both the `GenericHiveRecordCursorProvider` and `S3SelectRecordCursorProvider` instead of just the latter. It makes sense that if you _only_ want to test S3 select that you would do that, but the intention isn't immediately obvious- so maybe an inline comment about this binding _only_ the S3 select record cursor would be helpful here.

Reply:
Therefore added both RecordCursorProviders and explanatory comment in HiveTestUtils.

Reply:
Makes sense, thanks